contracts/presets/oracles/BaseOracle.sol

```diff
@@ -0,0 +1,53 @@
+pragma solidity >=0.8.0;
+
+import {IBaseTemplatesManager, Template} from "../../interfaces/IBaseTemplatesManager.sol";
+import {Initializable} from "oz-upgradeable/proxy/utils/Initializable.sol";
+import {IOracle} from "../../interfaces/oracles/IOracle.sol";
+
+/// SPDX-License-Identifier: GPL-3.0-or-later
+/// @title BaseOracle contract
+/// @dev A base oracle preset that provides basic but functional
+/// implementations for a set of Carrot oracle functions.
+/// @author Federico Luzzi - <federico.luzzi@protonmail.com>
+abstract contract BaseOracle is IOracle, Initializable {
+    bool public override finalized;
+    address public override kpiToken;
+    address internal oraclesManager;
+    uint128 internal templateVersion;
+    uint256 internal templateId;
+
+    error ZeroAddressKPIToken();
+    error InvalidTemplateId();
+    error InvalidTemplateVersion();
+
+    /// @dev Initializes the internal state of the preset, allowing it to provide a
```

Comment:
Remove the `a` here since it's referring to multiple implementations.

```suggestion
    /// @dev Initializes the internal state of the preset, allowing it to provide
```

Reply:
Addressed in 085321c61ca8d07a7d23bc66030e6a08ff25b02e